src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -11,6 +11,27 @@ namespace Microsoft {
 namespace Featurizer {
 namespace Featurizers {
 
+template<typename MatrixType>
+inline MatrixType& svd_flip(MatrixType& mat) {
```

Comment:
We should add comments that explain what this function is doing and why it is necessary.